erts/emulator/beam/erl_bif_binary.c

```diff
@@ -1037,10 +1037,9 @@ static int do_binary_match_compile(Eterm argument, Eterm *tag, Binary **binp)
 	words = 1;
 	comp_term = argument;
 	characters = binary_size(argument);
-    }
-
-    if (characters == 0) {
-	goto badarg;
+	if (characters == 0) {
+	    goto badarg;
+	}
```

Comment:
I think I probably need to add another check here to substitute this one

Reply:
Yep, some error cases that previously fell there won't anymore. Example: `binary:compile_pattern(abc)`

Reply:
Fixed in ammended commit